File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/contracts/MediaPlayer.java
package com.masterwok.simplevlcplayer.contracts;

import android.net.Uri;

import com.masterwok.simplevlcplayer.VlcMediaPlayer;

/**
 * This contract provides a way to interact with a media player.
 */
public interface MediaPlayer {

    void release();

    void play();

    void pause();

    void togglePlayback();

    void stop();

    void setMedia(Uri uri);

    void setSubtitle(Uri uri);

    void setCallback(VlcMediaPlayer.Callback callback);

    long getTime();

    void setTime(long time);

    long getLength();

    boolean isPlaying();

    interface Callback {
        void onPlayerOpening();

        void onPlayerSeekStateChange(boolean canSeek);

        void onPlayerPlaying();

        void onPlayerPaused();

        void onPlayerStopped();

        void onPlayerEndReached();

        void onPlayerError();

        void onPlayerTimeChange(long timeChanged);

        void onPlayerPositionChange(float positionChanged);
    }
}

File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/contracts/MediaPlayer.kt
package com.masterwok.simplevlcplayer.contracts

import android.net.Uri

/**
 * This contract provides a way to interact with a media player.
 */
interface MediaPlayer {

    var time: Long

    var callback: Callback?

    val length: Long

    val isPlaying: Boolean

    fun release()

    fun play()

    fun pause()

    fun stop()

    fun setMedia(uri: Uri?)

    fun setSubtitleUri(uri: Uri?)

    interface Callback {
        fun onPlayerOpening()

        fun onPlayerSeekStateChange(canSeek: Boolean)

        fun onPlayerPlaying()

        fun onPlayerPaused()

        fun onPlayerStopped()

        fun onPlayerEndReached()

        fun onPlayerError()

        fun onPlayerTimeChange(timeChanged: Long)

        fun onBuffering(buffering: Float)

        fun onPlayerPositionChanged(positionChanged: Float)

        fun onSubtitlesCleared()
    }
}